src/utils/healthcheck.ts

```diff
@@ -0,0 +1,182 @@
+import { CosmosClient } from "@azure/cosmos";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import {
+  common as azurestorageCommon,
+  createBlobService,
+  createFileService,
+  createQueueService,
+  createTableService
+} from "azure-storage";
+
+import * as A from "fp-ts/lib/Array";
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as RA from "fp-ts/lib/ReadonlyArray";
+import * as T from "fp-ts/lib/Task";
+import * as TE from "fp-ts/lib/TaskEither";
+
+import fetch from "node-fetch";
+
+import * as t from "io-ts";
+
+export type ProblemSource = "AzureCosmosDB" | "AzureStorage" | "Config" | "Url";
+
+// eslint-disable-next-line
+export type HealthProblem<S extends ProblemSource> = string & { __source: S };
+export type HealthCheck<
+  S extends ProblemSource = ProblemSource,
+  True = true
+> = TE.TaskEither<ReadonlyArray<HealthProblem<S>>, True>;
+
+// format and cast a problem message with its source
+const formatProblem = <S extends ProblemSource>(
+  source: S,
+  message: string
+): HealthProblem<S> => `${source}|${message}` as HealthProblem<S>;
+
+// utility to format an unknown error to an arry of HealthProblem
+const toHealthProblems = <S extends ProblemSource>(source: S) => (
+  e: unknown
+): ReadonlyArray<HealthProblem<S>> => [
+  formatProblem(source, E.toError(e).message)
+];
+
+/**
+ * Check application's configuration is correct
+ *
+ * @returns either true or an array of error messages
+ */
+export const checkConfigHealth: <IConfig>(
+  ConfigCodec: t.Type<IConfig>
+) => (value: unknown) => HealthCheck<"Config", IConfig> = <IConfig>(
+  ConfigCodec: t.Type<IConfig>
+) => (value: unknown): HealthCheck<"Config", IConfig> =>
```

Comment:
We may want to have cleaner signature:
```suggestion
export const checkConfigHealth = <IConfig>(
  ConfigCodec: t.Type<IConfig>
) => (value: unknown): HealthCheck<"Config", IConfig> =>
```

Reply:
Sure, thanks